script/build_catalog.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+# builds the OLM catalog index and pushes it to quay.io.
+#
+# To push to your own registry, override the IMG_REGISTRY_HOST , IMG_REGISTRY_ORG , OPERATOR_NAME and TAG env vars,
+# i.e:
+#   IMG_REGISTRY_HOST=quay.io IMG_REGISTRY_ORG=yourusername OPERATOR_NAME=authorino-operator TAG=latest ./script/build_catalog.sh
+#
+# REQUIREMENTS:
+#  * a valid login session to a container registry.
+#  * `docker`
+#  * `opm`
+#
+
+# Iterate over tag list i.e. latest 8a17c81d5e9f04545753e5501dddc4a0ac2c7e03
+IFS=' ' read -r -a tags <<< "$TAG"
+
+for tag in "${tags[@]}"
+do 
```

Comment:
This will detach the tags into separate builds (separate manifests). Other container images (operator and olm bundle) are built just once with multiple tags. Breaking with this pattern may affect automation that depend on the link from single manifest build.

Reply:
`opm index add` doesnt support build for multiple tags. Thats we have to iterate. buildah has the capability to build once with multiple tags which is not the case with opm

Reply:
I understand. As I mentioned before, I'd rather not break the link between tags built based on the exact same version of the code and build args, but to keep them all linked as part of the same manifest instead.

This not being possible here, I guess we'll have to wait until we get confirmation no automation currently depends on linked tags.

In the meantime, maybe we could build the catalogs for s390x and ppc64le separately from amd64 and arm64? We keep the current ones as they are based on buildah and add another step for s390x and ppc64le? WDYT?

Reply:
I am afraid thats not possible as its overwriting the images as we push it into the quay repo
<img width="1716" alt="Screenshot 2024-10-03 at 2 57 45 PM" src="https://github.com/user-attachments/assets/2bbc5593-70d6-4018-9cd3-2ea9ce4e3699">